mystore/app/models/spree/store_credit_decorator.rb

```diff
@@ -0,0 +1,11 @@
+Spree::StoreCredit.class_eval do
+
+  def expired?
+    expires_at.present? && expires_at < Time.current
+  end
+
+  def amount_remaining
+    expired? ? 0 : amount - amount_used - amount_authorized
```

Comment:
@rajneeshsharma9 
We should use `super` instead of `amount - amount_used - amount_authorized`